components/autocomplete/index.js

```diff
@@ -231,18 +232,26 @@ export class Autocomplete extends Component {
 		}
 	}
 
-	loadOptions( index ) {
-		this.props.completers[ index ].getOptions().then( ( options ) => {
-			const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + i } ) );
-			const filteredOptions = filterOptions( this.state.search, keyedOptions );
-			const selectedIndex = filteredOptions.length === this.state.filteredOptions.length ? this.state.selectedIndex : 0;
-			this.setState( {
-				[ 'options_' + index ]: keyedOptions,
-				filteredOptions,
-				selectedIndex,
+	/**
+	 * Load options for an autocompleter.
+	 *
+	 * @param {number} index The autocompleter index.
+	 * @param {string} query The query, if any.
+	 */
+	loadOptions( index, query ) {
+		this.props.completers[ index ]
+			.getOptions( query )
+			.then( ( options ) => {
+				const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + ( option.value.slug ? option.value.slug : i ) } ) );
```

Comment:
This line is excessively long. Implicit object arrow returns are hard to distinguish from arrow function body. Dense single lines ought be undesirable.

With some breathing room, and clear labeling of what we're constructing via variable assignment of the suffix.

```js
const keyedOptions = map( options, ( option, i ) => {
	const keySuffix = option.value.slug || i;

	return {
		...option,
		key: [ index, keySuffix ].join( '-' ),
	};
} );
```

Reply:
Aside: Existed previously, but as a reader, I'm not sure what the difference between `index` and `i` are here at a glance. Maybe we ought to name each autocompleter instead of using its index as an identifier.

Reply:
I renamed this variable `autocompleterIndex` to make it clearer.

Reply:
also, unfurled that long line as you suggested to make it easier to read